File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool solve(string & s){
        int i=0,j=s.size()-1;
        cout<<s;
        if(s.size()==1)return true;
        while(i<j){
            if(s[i]!=s[j]){
                // cout<<s[i]<<s[j]<<endl;
                return false;

            }
            i++;
            j--;
        }
        return true;
    }
    bool isPalindrome(string s) {
        string s1="";
        for( auto & x :s){
            if(isalnum(x)){
                x=tolower(x);
                s1+=x;
            }
        }
        cout<<s1;
        return solve(s1);
        
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        si=""
        for i in s:
            if i.isalnum():
                si+=i.lower()
        # print(si)
        # k=0
        i,j=0,len(si)-1
        while(i<j):
            if si[i]!=si[j]:
                return False
            i+=1
            j-=1
        return True